pkg/promotion/runner/builtin/argocd_updater.go

```diff
@@ -558,6 +546,25 @@ func (a *argocdUpdater) syncApplication(
 	return nil
 }
 
+// constructMessage construct a human-friendly message about what we're syncing to.
+//   - If there's exactly one source, we include its TargetRevision.
+//   - If there are multiple sources, we just report the count to avoid a noisy log.
+//   - If only the single Source field is set, we log its TargetRevision.
+//     Full details are always available on the Application object itself.
+func constructMessage(app *argocd.Application) string {
```

Comment:
Because of the large amount of code in this package, it would be preferable that this were a method of the `*argocdUpdater` type so that its relationship to other functionality in that same component is more obvious, so it isn't ever mistaken for something relevant to a different runner.

Also on account of the large volume of code in this package, please see about tweaking the name and the docstring for clarity and context. The current description still manages to be vague about what this is really used for.

Reply:
Please take a look now.